### PR TITLE
TaegisXDR v2

### PR DESCRIPTION
**Pull request by reedium**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Note: This is a continuation/rework of #27739 

This change primarily moves from the old Taegis Investigations v1 API to v2. The v1 API calls are being turned down in the near future and should not be used. Due to inconsistencies between the v1 and v2 APIs and pulling of data, users must move to the new API. For instance, some incidents will not be found in the v1 API but will in the v2 API.

The following changes were made during this PR:
- Deprecates the existing TaegisXDR integration and creates a new TaegisXDR v2 integration
- Updated the following commands to utilize the InvestigationsV2 API:
  - **taegis-fetch-comment**
  - **taegis-fetch-comments**
  - **taegis-fetch-incidents**
  - **taegis-fetch-investigation**
  - **taegis-update-comment**
  - **taegis-update-investigation**
- Added support for defining which type of "incident" to fetch from Taegis when using **taegis-fetch-incidents**. The following are supported:
  - **alerts**
  - **investigations (default)**
- Added support for overriding the fields returned from Taegis in most commands with the `fields` argument.
- Added support for defining the `tenant_id` when running most commands. This is most useful for MSPs and tenants that contain child tenants.
- Added support for the new US3 (Foxtrot) environment.
- Added 2 new commands:
  - **taegis-add-evidence-to-investigation** - Add alerts or events to an existing investigation
  - **taegis-create-sharelink**

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: The newer Investigations v2 API has different inputs and outputs. Additionally, some data is only available in the v2 API, such as newer investigations (incidents).
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/SecureWorks/ReleaseNotes/5_0_0.json
Comment (anas-yousef) [inline Packs/SecureWorks/ReleaseNotes/5_0_0.json:1]: No need for a breaking change json when deprecating an integration in a pack, you can delete this file.

File: Packs/SecureWorks/Integrations/TaegisXDRv2/Pipfile.lock
[the remark this answers is not in the document]
Comment (reedium) [reply Packs/SecureWorks/Integrations/TaegisXDRv2/Pipfile.lock:1]: Removed

File: Packs/SecureWorks/Integrations/TaegisXDRv2/Pipfile
[the remark this answers is not in the document]
Comment (reedium) [reply Packs/SecureWorks/Integrations/TaegisXDRv2/Pipfile:1]: Removed file

File: Packs/SecureWorks/Integrations/TaegisXDRv2/TaegisXDRv2.py [hunk 1]
```diff
@@ -0,0 +1,1764 @@
...
+
+    return results
+
+
+def fetch_comment_command(client: Client, env: str, args=None):
+    comment_id = args.get("id")
+    if not comment_id:
+        raise ValueError("Cannot fetch comment, missing comment_id")
+
+    fields: str = args.get("feilds") or """
```
[the remark this answers is not in the document]
Comment (reedium) [reply Packs/SecureWorks/Integrations/TaegisXDRv2/TaegisXDRv2.py]: Corrected Spelling

File: Packs/SecureWorks/Integrations/TaegisXDRv2/TaegisXDRv2.py [hunk 2]
```diff
@@ -0,0 +1,1764 @@
...
+    if not 0 < int(max_fetch) < 201:
+        raise ValueError("Max Fetch must be between 1 and 200")
+
+    if fetch_type not in ["alerts", "investigations"]:
+        raise ValueError("Incident Type is invalid. Supported types: ['alerts', 'investigations']")
+
+    last_run = demisto.getLastRun()
+    demisto.debug(f"Last Fetch Incident Run: {last_run}")
+    now = datetime.now()
+    start_time = str(now - timedelta(days=1))  # Default start if first ever run
```
[the remark this answers is not in the document]
Comment (reedium) [reply Packs/SecureWorks/Integrations/TaegisXDRv2/TaegisXDRv2.py:824]: Currently, the script doesn't utilize this (something we should look into for a future update) but changing to 1 day for clarity makes sense. I went ahead and changed this to 1 day.

**Review by anas-yousef — COMMENTED** [review]
Awesome job! I added one last comment, other than that, we are ready to have a demo! Can you please send me your availability via email for this week or next week so we can set it up? I believe we would need from 30 to 45 minutes to go over everything.

File: Packs/SecureWorks/Integrations/TaegisXDRv2/TaegisXDRv2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anas-yousef) [reply Packs/SecureWorks/Integrations/TaegisXDRv2/TaegisXDRv2.py:824]: The following code subtracts the time range from datetime.now() automatically
```suggestion
    start_time = str(dateparser.parser(first_fetch_interval))  # Default start if first ever run
```

**Comment by anas-yousef** [comment]
Hey @reedium,
I added one last fix that needs to be added and tested, so we could publish this PR, amazing job!

**Comment by reedium** [comment]
@anas-yousef I made a few changes:

* Made your suggested changes. However, I had to make a few slight tweaks for it to work (`dateparser.parse()` vs `dateparser.parser()`).
* Slight change to the environment names (e.g. `us1` -> `us1 (charlie)`) - This isn't important but I thought I'd add as we're moving to Taegis v2 and is a bit more clear to users
* Updated docker image python version for failing checks

All tests are showing passing and I believe this should be good.

**Review by anas-yousef — APPROVED** [review]

**Pull request merged**